src/Symfony/Component/Console/Output/ConsoleSectionOutput.php

```diff
@@ -168,6 +168,13 @@ public function addNewLineOfInputSubmit(): void
      */
     protected function doWrite(string $message, bool $newline)
     {
+        // if there is a new line at the end of message, but we do not add a new line, then simulate if without new line and add new line
+        // a lot of code relies on the newline bool to correctly format the output, it avoid a lot of new logic to handle both cases which are roughly the same
```

Comment:
what about just this?
`// Simulate newline behavior for consistent output formatting, avoiding extra logic`

Reply:
I used the first suggestion, thanks